lib/tee/filter.ex

```diff
@@ -1,22 +1,49 @@
 defmodule Membrane.Element.Tee.Filter do
   use Membrane.Element.Base.Filter
 
-  def_output_pads output: [
+  def_input_pad :input,
+    availability: :always,
+    mode: :pull,
+    demand_unit: :buffers,
     caps: :any
-  ]
 
-  def_input_pads input: [
-    caps: :any,
-    demand_unit: :buffers
-  ]
+  def_output_pad :output_static,
+    availability: :always,
+    mode: :pull,
+    caps: :any
+
+  def_output_pad :output_dynamic,
+    availability: :on_request,
+    mode: :push,
+    caps: :any
+
+  # TODO: extract pads mapping to separate function
+  @impl true
+  def handle_process(:input, %Membrane.Buffer{} = buffer, ctx, state) do
+    {{:ok,
+      ctx.pads
```

Comment:
1. It's not very readable, you should rather extract it to a variable
2. This piece of code repeats in 3 places, you can extract it to a private function and the variable might not be needed.

Reply:
Yup, wanted to do this anyway 😊

Reply:
done